src/ScriptCs.Core/ScriptLibraryComposer.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Common.Logging;
+using ScriptCs.Contracts;
+using System.IO;
+
+namespace ScriptCs
+{
+    public class ScriptLibraryComposer : IScriptLibraryComposer
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly IFilePreProcessor _preProcessor;
+        private readonly IPackageContainer _packageContainer;
+        private readonly IPackageAssemblyResolver _packageAssemblyResolver;
+        private readonly ILog _logger;
+
+        public ScriptLibraryComposer(IFileSystem fileSystem, IFilePreProcessor preProcessor, IPackageContainer packageContainer, IPackageAssemblyResolver packageAssemblyResolver, ILog logger)
+        {
+            Guard.AgainstNullArgument("fileSystem", fileSystem);
+            Guard.AgainstNullArgument("preProcessor", preProcessor);
+            Guard.AgainstNullArgument("packageContainer", packageContainer);
+            Guard.AgainstNullArgument("packageAssemblyResolver", packageAssemblyResolver);
+            Guard.AgainstNullArgument("logger", logger);
+
+            _fileSystem = fileSystem;
+            _preProcessor = preProcessor;
+            _packageContainer = packageContainer;
+            _packageAssemblyResolver = packageAssemblyResolver;
+            _logger = logger;
+        }
+
+        internal string GetMainScript(IPackageObject package)
+        {
+            var script =
+                package.GetContentFiles()
+                    .SingleOrDefault(f => f.EndsWith("main.csx", StringComparison.InvariantCultureIgnoreCase));
```

Comment:
Currently it will just end up ignored. It probably makes sense though to add warning. Another option is allowing 2 mains, but I don't currently think that makes a lot of sense